components/core/src/clp/ffi/ir_stream/utils.hpp

```diff
@@ -133,5 +192,88 @@ template <typename encoded_variable_t>
     }
     return succeeded;
 }
+
+template <IntegerType T>
+auto get_ones_complement(T int_val) -> T {
+    // Explicit cast to undo the implicit integer promotion
+    return static_cast<T>(~int_val);
+}
+
+template <
+        bool is_auto_generated_node,
+        int8_t one_byte_length_indicator_tag,
+        int8_t two_byte_length_indicator_tag,
+        int8_t four_byte_length_indicator_tag>
+auto encode_and_serialize_schema_tree_node_id(
+        SchemaTree::Node::id_t node_id,
+        std::vector<int8_t>& output_buf
+) -> bool {
+    auto size_dependent_encode_and_serialize_schema_tree_node_id
+            = [&output_buf]<SignedIntegerType encoded_node_id_t>(
```

Comment:
Shall we also capture node_id rather than passing it as an arg?

Reply:
sure